### PR TITLE
Improve applicant manager tests

### PR DESCRIPTION
**Pull request by dvacca**

### CODE REVIEW

File: onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java [hunk 1]
```diff
@@ -1,26 +1,81 @@
...
 
-  @BeforeTest
-  public void clearServer() {
-    server = null;
+  protected ApiIntegrationTest() {
+    if ( ! mockingEnabled() )
+    {
+      onfido = builder.apiToken(apiToken).regionEU().build();
+    }
   }
```
Comment (sergiomartins8) [inline onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java:30]: This bit doesn't have to be protected, I believe. Also coding style could be improved to be more java'ish styled  💅 

```suggestion
  public ApiIntegrationTest() {
    if (!mockingEnabled()) {
      onfido = builder.apiToken(apiToken).regionEU().build();
    }
  }
```

File: onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java [hunk 2]
```diff
@@ -1,26 +1,81 @@
...
 
+  @BeforeTest
+  private void setup() {}
+
```
Comment (sergiomartins8) [inline onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java:34]: If we're not using the `setup()`, we might as well remove it entirely 💄

File: onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java [hunk 3]
```diff
@@ -1,26 +1,81 @@
...
   @AfterTest
-  public void shutdownServer() throws IOException {
-    if (server != null) {
+  private void tearDown() throws IOException {
+    if ( mockingEnabled() && server != null )
+    {
       server.shutdown();
     }
   }
```
Comment (sergiomartins8) [inline onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java:41]: Coding style could be improved as well ✨ 

```suggestion
    if (mockingEnabled() && server != null) {
      server.shutdown();
    }
  }
```

File: onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java [hunk 4]
```diff
@@ -1,26 +1,81 @@
...
 
+  protected boolean mockingEnabled()
+  {
+    return ( apiToken == null );
+  }
```
Comment (sergiomartins8) [inline onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java:46]: Code style and naming could be improved too 💅 

```suggestion
  protected boolean isMockingEnabled() {
    return apiToken == null;
  }
```

File: onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java [hunk 5]
```diff
@@ -1,26 +1,81 @@
...
+  {
+    if ( mockingEnabled() )
+    {
+      server = new MockWebServer();
+      server.enqueue(new MockResponse().setBody(mockedResponse.toJson()));
+      server.start();
+
+      onfido = builder.apiToken("token").unknownApiUrl(server.url("/").toString()).build();
+    }
+  }
```
Comment (sergiomartins8) [inline onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java:58]: Code style to be improved here too. As well as on the remaining of the whole file 💄

**Comment by sergiomartins8** [comment]
Great job contributing to the tests of the library 💪 

A couple of suggestions, though:
1. These should def. be added to the pipeline before being merged
2. Code style could be improved a bit to a more java'ish approach 
3. Some help methods are floating around test files. I'd suggest that we extract those into helpers if you agree 💡

File: onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java (hunk 5 above)
Comment (dvacca) [reply onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java:55]: Are you just referring to:
- newline before closing curly bracket
- spaces in-between round brackets
Or is there any other code style improvement you're suggesting?

File: onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java (hunk 2 above)
Comment (dvacca) [reply onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java]: It's done

File: onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java (hunk 5 above)
Comment (dvacca) [reply onfido-java/src/test/java/com/onfido/integration/ApiIntegrationTest.java]: > * Code style could be improved a bit to a more java'ish approach
> * Some help methods are floating around test files. I'd suggest that we extract those into helpers if you agree 💡

Code style should be better now. Regarding the helper class, if you're referring to new methods inside `ApiIntegrationTest` class, that was the case at the beginning but it was making code in the test classes less readable (due to the call to the helper). Moreover in my understanding the aim `ApiIntegrationTest` is really to perform as an helper (all methods in the class looks fulfilling that role and the 3 last methods are expected to disappear as replaced by the new ones).

**Comment by dvacca** [comment]
> Great job contributing to the tests of the library 💪
> 
> A couple of suggestions, though:
> 
> 1. Code style could be improved a bit to a more java'ish approach
> 2. Some help methods are floating around test files. I'd suggest that we extract those into helpers if you agree 💡

Thanks for your review, code style should be better now. Regarding the helper class you're suggesting, are you thinking at the new methods inside ApiIntegrationTest class? It was like that at the beginning of my change but it was making code in the test classes less readable (due to the call to the helper). Moreover from my understanding the aim of `ApiIntegrationTest` is actually to be an helper class (all methods in the class looks fulfilling that role and moreover the 3 last methods are expected to disappear as replaced by the new ones), which is taking advantage of inheritance to have simpler code. What do you think on that?

**Review by DavidMealha — APPROVED** [review]

**Pull request merged**